content/developer/howtos/website_themes/theming.rst

```diff
@@ -0,0 +1,632 @@
+
+Theme module
+============
+
+Odoo comes with a default theme that provides minimal structure and layout. When you create a new
+theme, you are extending the default theme.
+
+Remember to add the directory containing your module to the `addons-path` command-line argument
+when running Odoo in your development environment.
+
+Technical naming
+----------------
+
+The first step is to create a new directory.
+
+.. code-block:: xml
+
+   website_airproof
+
+.. note::
+   Prefix it with `website_` and use only lowercase ASCII alphanumeric characters and underscores.
+
+File structure
+--------------
+
+Themes are packaged like any Odoo module. Even if you are designing a basic website, you need to
+package its theme like a module.
+
+::
+
+    website_airproof
+    ├── data
+    ├── i18n
+    ├── lib
+    ├── static
+    │   ├── description
+    │   ├── fonts
+    │   ├── image_shapes // Shapes for images
+    │   ├── shapes // Shapes for background
+    │   └── src
+    │       ├── img
+    │       │   ├── content // For those used in the pages of your website
+    │       │   └── wbuilder // For those used in the builder
+    │       ├── js
+    │       ├── scss
+    │       └── snippets // custom snippets
+    ├── views
+    __init__.py
+    __manifest__.py
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 20 80
+
+   * - Folder
+     - Description
+   * - data
+     - Presets, menus, pages, images, shapes, … (`*.xml`)
+   * - i18n
+     - Translations (`*.po`, `*.pot`)
+   * - lib
+     - External libraries (`*.js`)
+   * - static
+     - Custom assets (`*.jpg`, `*.gif`, `*.png`, `*.svg`, `*.pdf`, `*.scss`, `*.js`)
+   * - views
+     - Custom views and templates (`*.xml`)
+
+Initialization
+--------------
+
+An Odoo module is also a Python package with a :file:`__init__.py` file containing import
+instructions for various Python files in the module. This file can remain empty for now.
+
+Declaration
+-----------
+
+An Odoo module is declared by its manifest file. This file declares a Python package as an Odoo
+module and specifies the module's metadata. It must at least contain the `name` field, which is
+always required. It usually contains much more information.
+
+.. code-block:: python
+   :caption: ``/website_airproof/__manifest__.py``
+
+   {
+      'name': 'Airproof Theme',
+      'description': '...',
+      'category': 'Website/Theme',
+      'version': '15.0.0',
+      'author': '...',
+      'license': '...',
+      'depends': ['website'],
+      'data': [
+	     # ...
+      ],
+      'assets': {
+	     # ...
+      },
+   }
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 20 80
+
+   * - Field
+     - Description
+   * - name
+     - Human-readable name of the module (required)
+   * - description
+     - Extended description of the module, in `reStructuredText
+       <https://en.wikipedia.org/wiki/ReStructuredText>`_
+   * - category
+     - Classification category within Odoo
```

Comment:
It's worth noting that this is what makes a theme be considered as a theme by Odoo.